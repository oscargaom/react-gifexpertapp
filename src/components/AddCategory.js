import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleChange = (e) => {
        setInputValue(e.target.value);
        //console.log(`handleChange take place`);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        //console.log(`handleSubmit take place with value: "${inputValue}"`);

        if (inputValue.trim().length > 0 ) {
            setCategories(cat => [inputValue, ...cat]);
            //console.log(`setCategories called with value: "${inputValue}"`);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
