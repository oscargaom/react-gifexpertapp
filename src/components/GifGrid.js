// import React, { useState, useEffect } from 'react'
import React from 'react'
import { useFecthGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';
// import { getGifs } from '../helpers/GetGifs';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([]);

    // getGifs();
    const {data: images, loading} = useFecthGifs(category);
    // console.log(data);
    // console.log(loading);


    return (
        <>
            <h3 className="animate__animated animate__fadein">{category}</h3>
            {
                loading && <p className="animate__animated animate__flash">Loading...</p>
            }
            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id} 
                            {...img} 
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}