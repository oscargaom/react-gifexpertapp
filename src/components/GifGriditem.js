import React from 'react'

export const GifGriditem = ({title, url}) => {


    return (
        <div className="card animate__animated animate__fadein">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
