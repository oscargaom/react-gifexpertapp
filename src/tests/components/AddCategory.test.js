import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas unitarias del componente <AddCategory/>', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });
    
    test('Validar que el componente sea valido', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Validar el evento handleChange', () => {
        const input = wrapper.find("input");
        const value = "Hola mundo!!!";
        
        input.simulate("change", { target: { value } });
        
        const p = wrapper.find("p");
        
        expect(p.text()).toBe(value);
    })

    test('Validar el evento handleSubmit vacío', () => {
        const form = wrapper.find("form");

        form.simulate("submit", {
            preventDefault: () => {}
        });

        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('Validar el evento handleSubmit con datos', () => {
        const input = wrapper.find("input");
        const value = "Submit test";
        
        input.simulate("change", { target: { value } });
        
        const form = wrapper.find("form");

        form.simulate("submit", {
            preventDefault: () => {}
        });
        
        expect(setCategories).toHaveBeenCalled();
    })

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const input = wrapper.find("input");
        const value = "Prueba número 5";
        
        input.simulate("change", {
            target: {
                value
            }
        });

        const form = wrapper.find("form");

        form.simulate("submit", {
            preventDefault: () => {}
        });

        
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toBeCalledWith( expect.any(Function) );
        expect(wrapper.find('input').prop('value')).toBe("");
    })
    
    
})
