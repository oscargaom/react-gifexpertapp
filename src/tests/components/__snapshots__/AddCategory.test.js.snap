// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias del componente <AddCategory/> Validar que el componente sea valido 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
